### PR TITLE
Add `fallback` Parser method

### PR DESCRIPTION
**Pull request by leonardfactory**
Parser `else` method, equals to `or(Parsimmon.of(…))`. #105 

```js
var parser = P.string('a').else('x');

parser.parse('a').value; // => 'a'
parser.parse('').value;   // => 'x'
```

Includes test & doc :smile: 

Talking about the name: It seems to me that `default` may be a nice one, since this method is just providing a value if parser fails, and does *not* consume characters from the stream

```js
var parser = P.letters.default('empty');
```

### CODE REVIEW

**Comment by wavebeem** [comment]
Oh, I think I really like the name `default`. Let's go with that.

File: src/parsimmon.js [hunk 1]
```diff
@@ -420,6 +420,13 @@
     });
   };
 
+  _.else = function(result) {
+    return this.or(succeed(result));
+  };
+
+  //- ES3 compatibility
+  _.else_ = _.else;
+
```
Comment (wavebeem) [inline src/parsimmon.js]: For ES3 compatibility we'll need this as `_["default"] = function...` and then `_.default_ = _["default"]`

**Comment by wavebeem** [comment]
Just realized `default` might be confusing to people learning ES6 modules, as this is NOT the default export from Parsimmon.

File: src/parsimmon.js (hunk 1 above)
Comment (leonardfactory) [inline src/parsimmon.js]: Oh yup, what a mess! Fixing it now

**Comment by leonardfactory** [comment]
You're right, however `Parsimmon.default` is not exported, just`Parser.default`?
Maybe `defaultsTo` or sticking with `else`?

**Comment by wavebeem** [comment]
`.fallback` has the advantage of being pretty clear and not being a reserved word.

**Comment by leonardfactory** [comment]
Nice! 👍

**Comment by wavebeem** [comment]
Not my idea, I've been bikeshedding with other people outside of GitHub 😄

**Comment by wavebeem** [comment]
Would you mind adding this new feature to `CHANGELOG.md` under the `version ?` section? Everything else looks good to go. Thanks again for the test and docs.

**Comment by leonardfactory** [comment]
May I use this PR to add changelog even for `P.parse`old PR? i.e.

```
* Documentation for `P.formatError`, `P.parse`
```

**Comment by wavebeem** [comment]
Sure! That'd be great. Just trying to make sure I don't forget to document the new stuff in the CHANGELOG.

**Comment by wavebeem** [comment]
Thanks again for your work on this!

**Pull request merged**